python/cudf/dataframe/string.py

```diff
@@ -127,9 +127,72 @@ def cat(self, others=None, sep=None, na_rep=None):
             (same type as caller) of str dtype is returned.
         """
         from cudf.dataframe import Series, Index
+
         if isinstance(others, (Series, Index)):
+            '''
+            If others is just another Series/Index,
+            great go ahead with concatenation
+            '''
             assert others.dtype == np.dtype('object')
             others = others.data
+        elif utils.is_list_like(others) and others:
+            '''
+            If others is a list-like object (in our case lists & tuples)
+            just another Series/Index, great go ahead with concatenation.
+            '''
+
+            '''
+            Picking first element and checking if it really adheres to
+            list like conditions, if not we don't have the
+            necessity to scan the rest of the list.
+            '''
+            first = others[0]
+
+            if (utils.is_list_like(first) or
+                isinstance(first, (Series, Index, pd.Series, pd.Index))) \
+                    and all((
+                            utils.is_list_like(x) or
+                            isinstance(x, (Series, Index, pd.Series, pd.Index))
+                            )
+                            for x in others):
+                '''
+                Internal elements in others list should also be
+                list-like and not a regular string/byte
+                '''
+
+                first = None
+                for frame in others:
+                    if not isinstance(frame, (Series, Index)):
+                        '''
+                        Make sure all inputs to .cat function call
+                        are of type nvstrings so creating a Series object.
+                        '''
+                        frame = Series(frame, dtype='str')
+
+                    if (first is None):
+                        '''
+                        extracting nvstrings pointer since
+                        `frame` is of type Series/Index and
+                        first isn't yet initialized.
+                        '''
+                        first = frame.data
+                    else:
+                        assert frame.dtype == np.dtype('object')
+                        frame = frame.data
+                        first = first.cat(frame, sep=sep, na_rep=na_rep)
+
+                others = first
+            elif (not utils.is_list_like(first)) and \
+                    all(not utils.is_list_like(x) for x in others):
```

Comment:
Again, don't want to iterate over all the elements serially here as this could be quite large if composed of strings.

Reply:
Agree, removed it here as well.